trackbasket_be/models/krogerservice.py

```diff
@@ -1,115 +1,117 @@
 from flask_restful import Resource, reqparse, request
 from .at_risk_user import AtRiskUser
-import requests
-import os
-import json
+import requests, os, json
 
-proxyDict = {
-              "http"  : os.environ.get('FIXIE_URL', ''),
-              "https" : os.environ.get('FIXIE_URL', '')
-            }
+proxyDict = { "http"  : os.environ.get('FIXIE_URL', ''), "https" : os.environ.get('FIXIE_URL', '') }
 
 class Krogerservice:
 
   @classmethod
-  def return_token(cls):
+  def closest_store(cls, zipcode):
+    response  = cls.__store_api_request(zipcode)
+    store = cls.__extract_kroger_store(response)
+    if store is None:
+      return { 'error': 'no store found for this zipcode' }
+    return  { 'location_id': store['locationId'], 'name': store['chain'],'address': store['address']['addressLine1'],
+            'city': store['address']['city'],'state': store['address']['state'], 'zipcode': store['address']['zipCode'],
+            'latitude': store['geolocation']['latitude'],'longitude': store['geolocation']['longitude'] }
+  
+  @classmethod
+  def item_search(cls, term, at_risk_user_id):
+    nearest_store_id = cls.__get_store_id(at_risk_user_id)
+    if term == '':
+      return 'error'
+    json_items = cls.__item_api_request(term, nearest_store_id)
+    if json_items == []:
+      return 'error'
+    items = cls.__extract_items(json_items)
+    return { 'data': { 'id': 'items', 'attributes':  items } }
+
+  @classmethod
+  def __return_token(cls):
     try:
       access_token
     except NameError:
-      return Krogerservice.refresh_token()
+      return cls.__refresh_token()
     else:
       return access_token
 
   @classmethod
-  def refresh_token(cls):
-    client = os.environ.get("CLIENT")
-    authorization = 'Basic {}'.format(client)
-    data = { 'grant_type':'client_credentials', 'scope': 'product.compact' }
-    headers = {'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': authorization}
-    response = requests.post('https://api.kroger.com/v1/connect/oauth2/token', headers=headers, data=data)
-    parsed_response = response.json()
-    if response.status_code != 200:
-      return  { 'error': 'could not refresh token' }
-    access_token = parsed_response['access_token']
+  def __refresh_token(cls):
+    global access_token  
+    response = cls.__token_api_request()
+    access_token = response.json()['access_token']
     return access_token
 
   @classmethod
-  def closest_store(cls, zipcode):
-    token = Krogerservice.refresh_token()
-    number_results = 5
+  def __find_image(cls, item):
+    if 'images' in item.keys():
+      for image in item['images']:
+        if (('perspective' in image.keys()) and image['perspective'] == 'front'):
+          return next((size['url'] for size in image['sizes'] if size['size'] == 'medium'), None)
```

Comment:
finds and returns the first item in the iterable that matches the given condition